Client/Frontend/Reader/ReaderMode.swift

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import WebKit
+
+enum ReaderModeState: String {
+    case Available = "Available"
+    case Unavailable = "Unavailable"
+    case Active = "Active"
+}
+
+/// Delegate that contains callbacks that we have added on top of the built-in WKWebViewDelegate
+protocol ReaderModeDelegate {
+    func readerMode(readerMode: ReaderMode, didChangeReaderModeState state: ReaderModeState, forBrowser browser: Browser)
+}
+
+class ReaderMode: BrowserHelper {
+    var delegate: ReaderModeDelegate?
+
+    private let browser: Browser
```

Comment:
...but there is one here as mentioned in IRC. ReaderMode -> Browser -> helpers -> ReaderMode.